cibyl/plugins/openstack/test_collection.py

```diff
@@ -0,0 +1,56 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+
+from cibyl.models.model import Model
+
+# pylint: disable=no-member
+
+
+class TestCollection(Model):
+    """ Model for a collection of tests run in an Openstack deployment."""
+
+    API = {
+        'tests': {
+            'arguments': []
+        },
+        'setup': {
+            'attr_type': str,
+            'arguments': []
+        }
+    }
+
+    def __init__(self, tests: set = None, setup: str = None):
+        if tests is None:
+            tests = set()
+        super().__init__({'tests': tests, 'setup': setup})
+
+    def merge(self, other):
+        """Merge the information of two TestCollection objects representing the
+        same collection of tests.
+
+        :param other: The TestCollection object to merge
+        :type other: :class:`.TestCollection`
+        """
+        if not self.setup.value:
+            self.setup.value = other.setup.value
+        if other.tests.value:
+            other_tests = other.tests.value
+            if self.tests.value:
+                own_tests = self.tests.value
+                all_tests = own_tests.union(other_tests)
+                self.tests.value = all_tests
+            else:
+                self.tests.value = other_tests
```

Comment:
You could avoid all these conditions by just doing the 'union' branch from the get go. The result is going to be the same if one of the two sides is empty.

Reply:
You're right, this could be quite simpler, updated thanks!